packages/core/admin/admin/src/components/Filters.tsx

```diff
@@ -0,0 +1,573 @@
+import { useIntl } from 'react-intl';
+
+import {
+  BASE_FILTERS,
+  CONTAINS_FILTERS,
+  FilterOption,
+  IS_SENSITIVE_FILTERS,
+  NUMERIC_FILTERS,
+  STRING_PARSE_FILTERS,
+} from '../constants/filters';
+import { MainField } from '../content-manager/utils/attributes';
+import { useControllableState } from '../hooks/useControllableState';
+
+import { createContext } from './Context';
+import { Form, InputProps } from './Form';
+import { InputRenderer } from './FormInputs/Renderer';
+
+/* -------------------------------------------------------------------------------------------------
+ * Root
+ * -----------------------------------------------------------------------------------------------*/
+
+interface FilterFormData {
+  name: string;
+  filter: string;
+  value?: string;
+}
+
+interface FitlersContextValue {
+  disabled: boolean;
+  onChange: (data: FilterFormData) => void;
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  setTriggerNode: (node: HTMLButtonElement | null) => void;
+  triggerNode: HTMLButtonElement | null;
+  options: Filters.Filter[];
+}
+
+const [FiltersProvider, useFilters] = createContext<FitlersContextValue>('Filters');
+
+interface RootProps
+  extends Partial<Pick<FitlersContextValue, 'disabled' | 'onChange' | 'options' | 'open'>> {
+  children: React.ReactNode;
+  defaultOpen?: boolean;
+  onOpenChange?: (open: boolean) => void;
+}
+
+const Root = ({
+  children,
+  disabled = false,
+  onChange,
+  onOpenChange,
+  open: openProp,
+  defaultOpen,
+  options = [],
+}: RootProps) => {
+  const [triggerNode, setTriggerNode] = React.useState<FitlersContextValue['triggerNode']>(null);
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  const handleChange = (data: FilterFormData) => {
+    if (onChange) {
+      onChange(data);
+    }
+  };
+
+  return (
+    <FiltersProvider
+      disabled={disabled}
+      onChange={handleChange}
+      open={open}
+      options={options}
+      setOpen={setOpen}
+      setTriggerNode={setTriggerNode}
+      triggerNode={triggerNode}
+    >
+      {children}
+    </FiltersProvider>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * Trigger
+ * -----------------------------------------------------------------------------------------------*/
+
+const Trigger = React.forwardRef<HTMLButtonElement, Filters.TriggerProps>(
+  ({ label }, forwardedRef) => {
+    const { formatMessage } = useIntl();
+    const { setTriggerNode, setOpen } = useFilters('Trigger', ({ setTriggerNode, setOpen }) => ({
+      setTriggerNode,
+      setOpen,
+    }));
+    const disabled = useFilters('Trigger', ({ disabled }) => disabled);
+    const open = useFilters('Trigger', ({ open }) => open);
+
+    const composedRefs = useComposedRefs(forwardedRef, setTriggerNode);
+
+    const handleClick = () => setOpen(!open);
```

Comment:
I am just wondering if we really need this variable

Reply:
I think that's a flavour of coding, typically I declare event handlers on their own lines to make the JSX more readable 😄